collectors_app.php

```diff
@@ -21,68 +21,108 @@
+    $db = new PDO(
+        'mysql:host=DB;dbname=collection',
+        'root',
+        'password'
+    );
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    return $db;
 }
-//----------------------------------------------------------------------------------------------------------------------form $_GET data
-$test = '';
-if ($_GET) {
-    $test = "{$_GET['size']} {$_GET['pattern']} {$_GET['color']}";
-
-    //------------------------------------------------------------------------------------------------------------------move form to DB
-    $query = $db->prepare("INSERT INTO `socks` (`size`, `pattern`, `color`) VALUES (:size, :pattern, :color)");
-    $result = $query->execute([
-        'size' => $_GET['size'],
-        'pattern' => $_GET['pattern'],
-        'color' => $_GET['color']
-    ]);
+//----------------------------------------------------------------------------------------------------------------------get table function
+function getTable($table, $db) {
+    $query = $db->prepare("SELECT `{$table}` FROM `{$table}s`");
+    $result = $query->execute();
+    if ($result) {
+        return $query->fetchAll();
+    } else {
+        echo 'not working'; //maybe return not working
+    }
+}
+//----------------------------------------------------------------------------------------------------------------------creating html options for select dropdown
+function dropDownOptions($rows) {
+    $options= '';
+    foreach($rows as $row) {
+        foreach ($row as $key => $value) {
+            $options .= "<option value=\"{$value}\">{$value}</option>";
+        }
+    }
+    return $options;
 }
+//----------------------------------------------------------------------------------------------------------return full html dropdown - select, options, /select
+function createDropdown($name) {
 
+    $start = "<select name=\"{$name}\" id=\"{name}\"><option selected=\"selected\">$name</option>";
+    $options = dropDownOptions(getTable($name, connectDB()));
+    $end = '</select><br><br>';
+    return $start . $options . $end;
+}
+function createSocksDiv($socksArr) {
+    $socksStr= '';
+    foreach($socksArr as $sock) {
+        $socksStr .= "<div class=\"sock\">
+        <p>Size: {$sock['size']}</p>
+        <p>Pattern: {$sock['pattern']}</p>
+        <p>Color: {$sock['color']}</p>
+        </div>";
+    }
+    return $socksStr;
+}
+function displaySocksCollection($db) {
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    $query = $db->prepare('SELECT `id`, `size`, `pattern`, `color` FROM `socks`');
+    $result = $query->execute();
+    if ($result) {
+        $socksArr = $query->fetchAll();
+    } else {
+        echo 'not working';
+    }
+    return createSocksDiv($socksArr);
+}
 ?>
 
 <!DOCTYPE html>
 <html lang="en">
 <head>
     <meta charset="UTF-8">
     <title>collectors_app</title>
-    <link rel="stylesheet" href="styles.css">
+    <style>
+        <?php include 'styles.css'; ?>
```

Comment:
What is this? Just use a normal link tag plz

Reply:
Is this in reference to the <style> tag? When I used <link> tags in the head, it didn't work. So I used the <style> tags